compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -7722,4 +7722,19 @@ private FieldInfo(List<BType> types, boolean required, boolean readonly) {
             this.readonly = readonly;
         }
     }
+
+    /**
+     * Check whether a record type has required fields.
+     *
+     * @param recordType Record type.
+     * @return true if the record type has required fields; false otherwise.
+     */
+    public boolean checkRecordTypeForRequiredFields(BRecordType recordType) {
```

Comment:
I think we can have this in the SemanticAnalyzer itself. Shall we also rename this to something like
```suggestion
    public boolean hasRequiredFields(BRecordType recordType) {
```